### PR TITLE
Slice GPU batched kernel

### PR DESCRIPTION
**Pull request by jantonguirao**

### CODE REVIEW

File: dali/kernels/slice/slice_gpu.cuh [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  size_t block_end = blocks[blockIdx.x].offset + blocks[blockIdx.x].size;
+  auto &sample = samples[sampleIdx];
+  auto *out = static_cast<OutputType*>(sample.out);
+  auto *in = static_cast<const InputType*>(sample.in);
+
+  for (; offset < block_end; offset += blockDim.x) {
+    std::size_t idx = offset;
+    std::size_t out_idx = idx;
+    std::size_t in_idx = 0;
+    for (std::size_t d = 0; d < Dims; d++) {
```
Comment (mzient) [inline dali/kernels/slice/slice_gpu.cuh]:
```suggestion
    for (std::size_t d = 0; d < Dims-1; d++) {
```

File: dali/kernels/slice/slice_gpu.cuh [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      unsigned int i_d = idx / sample.out_strides[d];
+      idx = idx % sample.out_strides[d];
+      in_idx += i_d * sample.in_strides[d];
+    }
```
Comment (mzient) [inline dali/kernels/slice/slice_gpu.cuh]:
```suggestion
    }
    in_idx += idx;
```

File: dali/kernels/slice/slice_gpu.cuh (hunk 1 above)
Comment (jantonguirao) [reply dali/kernels/slice/slice_gpu.cuh]: done

**Comment by jantonguirao** [comment]
!build

File: dali/pipeline/operators/crop/new_crop.h [hunk 3]
```diff
@@ -46,6 +47,7 @@ class NewCrop : public Operator<Backend>, protected CropAttr {
   DALIDataType input_type_;
   DALIDataType output_type_;
   std::size_t C_;
+  kernels::ScratchpadAllocator scratch_alloc_;
```
[the remark this answers is not in the document]
Comment (jantonguirao) [reply dali/pipeline/operators/crop/new_crop.h]: done (See SliceBase)

**Comment by jantonguirao** [comment]
!build

**Pull request merged**